app/models/workers_scaling/workers_scaling_algorithm_runner.rb

```diff
@@ -0,0 +1,47 @@
+##
+# Scheduler to run WorkersScalingAlgorithm decision loop periodically
+class WorkersScalingAlgorithmRunner
+
+  ##
+  # Expects:
+  # * instance of Experiment
+  # * instance of WorkersScalingAlgorithm for given Experiment
+  # * interval between decision loop executions in seconds
+  def initialize(experiment, algorithm, interval)
+    @experiment = experiment
+    @algorithm = algorithm
+    @interval = interval
+    @next_execution_time = Time.now + interval
+  end
+
+  ##
+  # Initializes WorkersScalingAlgorithm, then starts its decision loop
+  # Stops when there is no next execution time set
+  def start
```

Comment:
Dodaj informację w docu, jak będzie z tym problem to się zmieni. @next_execution_time jest teraz modyfikowana w dwóch wątkach.